src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -104,7 +106,16 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
-    const signedUrl = await this.prepareWebSocketUrl();
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    let endpointUrl = !reconnecting ? this.configuration.endpointUrl : null;
+    if (endpointUrl === null) {
+      const endpoint = await this.configuration.chimeClient.send(
```

Comment:
So if `this.configuration.endpointUrl` is `undefined` while connecting at the start, then here, do you want to set the `this.configuration.endpointUrl = endpoint.Endpoint.Url` ? The docstring says so:
"endpointUrl is deprecated as internally it is set on connect via chimeClient if null.."

Reply:
I don't think we want to expose it in the configuration as that should not exist and just be an internal thing

Reply:
Got it, so should the docstring contain why builders or when rather they can provide their own URL? Also, what is the impact when re-connecting if the builder provided URL at the start is different than the internally resolved one?

Reply:
The builder should never set this URL directly.  If they do, and in all examples or demos they do it by calling getMessagingSessionEndpoint and then passing it in.  If they don't get the URL by calling getMessagingSessionEndpoint then there is no guarantee it will connect to anything and thus is broken if they have a hardcoded value.

This calling of `getMessagingSessionEndpoint` is what will be done internally via the PR, so this field is really strictly deprecated and should not be used as it has no purpose now.  There is no real use case to get the URL from the configuration object, which is why it doesn't really make sense to do extra work to keep it up to date.

In retrospect, the original implementation should have never exposed this URL and done the resolving on connect/reconnect like this PR implements.

Reply:
Got it thanks for the explanation!